libp2p/nameresolving/dnsresolver.nim

```diff
@@ -110,7 +114,10 @@ method resolveIp*(
       try:
         let resp = await fut
         for answer in resp.answers:
-          resolvedAddresses.incl(answer.toString())
+          resolvedAddresses.incl(
+            try: answer.toString()
+            except Exception as exc: raise newException(CatchableError, exc.msg)
```

Comment:
Same as in previous comment. Rule of thumb is usually to remap to `Defect` if not able to handle in place and specially when remapping `Exception` - there is simply no way of knowing what type of error we're handling and if it's possible to recover from it at all.